htdocs/web_portal/controllers/downtime/downtime_utils.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*_____________________________________________________________________________
+ *=============================================================================
+ * File: downtime_utils.php
+ * Author: GOCDB DEV TEAM, STFC.
+ * Description: Helper functions which can be re-used while adding
+ *              or editing a downtime.
+ *
+ * License information
+ *
+ * Copyright 2013 STFC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ /*====================================================== */
+require_once __DIR__ . '/../../../../lib/Gocdb_Services/Factory.php';
+
+use DateTime;
+use DateTimeZone;
+
+/**
+ * Sorts the impacted IDs into impacted services and impacted endpoints.
+ *
+ * @param array $impactedIDs An array of `impactedIDs` which user has selected.
+ *
+ * @return array An array containing
+ *               `$siteLevelDetails` and `serviceWithEndpoints`.
+ */
+function endpointToServiceMapping($impactedIDs)
+{
+    $siteLevelDetails = [];
+    $serviceWithEndpoints = [];
+
+    /**
+     * For each impacted ID,
+     * sort between endpoints and services using the prepended letter.
```

Comment:
Would this NOT be enough of why we are using str_replace, @gregcorbett ?

Reply:
I think some explanation `$siteNumber`, `$parentService`, `$idType` would be helpful.

Reply:
Sure. Addressed in 2fce820b84488cbc2ba64ea363c122c25f05a2b7